src/flask/debughelpers.py

```diff
@@ -122,12 +122,17 @@ def _dump_loader_info(loader):
 
 
 def explain_template_loading_attempts(app, template, attempts):
-    """This should help developers understand what failed"""
+    """This should help developers understand what failed
+
+    .. versionchanged:: 1.0.x
+       Refactored codes.
+    """
     info = ['Locating template "%s":' % template]
     total_found = 0
     blueprint = None
     reqctx = _request_ctx_stack.top
-    if reqctx is not None and reqctx.request.blueprint is not None:
+
+    if reqctx and reqctx.request.blueprint is not None:
```

Comment:
This line should not be changed. `if thing` and `if thing is not None` have different meanings. The later is also faster because it avoids bool conversion.